docs/error-type.md

```diff
@@ -0,0 +1,94 @@
+# Solidity `error` type
+
+## Motivation
+
+We want to support typed errors in a syntactically equivalent and convenient way like
+the programmer can define `event` types.
+
+Example:
+
+```solidity
+contract C {
+	event HighestBidIncreased(address bidder, uint amount);
+	error BiddingFailed(string reason);
```

Comment:
I think we should also allow error definitions at file level.

Reply:
agreed!